### ISSUE TITLE
Make and ./consul_exporter not working

### ISSUE THREAD

**Issue by MobinAnsari** [prometheus/consul_exporter#71]
Hi Team,

I have download the zip file of consul_exporter from github uploaded to our consul server and unzipped it.
But when I try to run it by using make command. I am getting below error:
=============================
/bin/sh: go: command not found
>> formatting code
/bin/sh: go: command not found
make: *** [format] Error 127
==============================
-bash: ./consul_exporter: No such file or directory

I don't want to deploy the exporter using docker. I want to deploy it similar to the default node_exporter that is provided by prometheus and I want to run it through ./consul_exporter.

Please provide the correct way to deploy consul exporter on consul server.

Awaiting for your immediate response.

Regards,
Mobin Ansari

**Comment by linuxrebel** [comment]
Having similar issues as well  the return when compiling on Linux with go 1.8 and 1.9.2 is
>> formatting code
can't load package: package _/usr/src/consul_exporter: cannot find package "_/usr/src/consul_exporter" in any of:
        /usr/local/go/src/_/usr/src/consul_exporter (from $GOROOT)
        /go/src/_/usr/src/consul_exporter (from $GOPATH)
Makefile:35: recipe for target 'format' failed
make: *** [format] Error 1

I've now tried this on 3 systems that otherwise can compile at least a simple "hello world" app.

**Comment by amang017** [comment]
@grobie 
I followed your instruction. But its giving me this error now!!
![image](https://user-images.githubusercontent.com/30700440/33036805-2e966146-cdf5-11e7-9eaf-d82f3b696381.png)
thanks in advance

**Comment by grobie** [comment]
Hm, I guess the expectation is that you set up the go dev environment including adding `$GOPATH/bin` to your `$PATH`.

Try the following:

```bash
export PATH="$GOPATH/bin:$PATH"
make
```

**Comment by amang017** [comment]
Thats already done actually. The first error I recieved after running **make** command was **.../go command not found**. So I added `$GOPATH/bin` to `$PATH` variable. So I am assuming its trying to run some **promu** command (which is not explained in official reference-doc btw), when I execute **make** command.

**Comment by grobie** [comment]
Yes is, and `promu` gets installed automatically by make. You can do that manuall with `make promu`. It's just calling the standard `go get` to install a go package including a binary. The Makefile assumes `promu` to be installed as `PROMU := $(GOPATH)/bin/promu`. Make sure that $GOPATH is set correctly when calling make.

**Comment by amang017** [comment]
OK. 
**1.** So after I ran `make `command, I see a **promu** file created inside `$GOPATH/bin` folder. (FYI my GOPATH=/home/cloud-user/go -> that is my go workspace). 
**2.** I also see a **promu** directory being created along side my consul_exporter folder, I will come back to this later. 
Now for **1**, so I created env variable `PROMU=$GOPATH/src/bin/` in **.bash_profile** file & did `source ~/.bash_profile`. So this means, promu command should work anywhere in the console. So I type `promu` in terminal and get the error msg below:
![image](https://user-images.githubusercontent.com/30700440/33081685-827e8fee-cea0-11e7-8211-3789e114f52a.png)
Now I navigated to the folder where my **consul_exporter** is extracted, and ran `promu `command again, but this time terminal recognizes it (but doesnt recognizes elsewhere I dont know why) and gives msg below:
![image](https://user-images.githubusercontent.com/30700440/33081959-342c23f0-cea1-11e7-978e-e115c50a2936.png)
But, when I ran `make `command here, it gives same **/bin/promu: Command not found** error: 
![image](https://user-images.githubusercontent.com/30700440/33082202-d23a6af2-cea1-11e7-8c0c-dafb573b46e4.png)
Now for **2**, I navigated to the **promu** folder and see some files along with **Makefile** (so I am guessing promu installation folder is here)
![image](https://user-images.githubusercontent.com/30700440/33082701-03e87278-cea3-11e7-9d18-440840796c05.png)
I ran `make `command here and received following errors:
![image](https://user-images.githubusercontent.com/30700440/33082769-2fe25c40-cea3-11e7-9c09-7a83d46c2d70.png)
:(
So finally, I modified my $PROMU env variable with this location (i.e. location of promu installation I suppose `/home/cloud-user/go/src/github.com/prometheus/promu/`), but when I type promu again, I am getting same error: /bin/promu: Command not found.

So I have no clue now. I am trying to be as detailed as possible and please let me know if I am missing anything. Thanks for your help!

**Comment by gargamile** [comment]
@amang017  
I was seeing the same problem as you and when running `make -d` from the $GOPATH/src/github.com/prometheus/consul_exporter/ directory, I returned the error:

```
.....
make: /bin/promu: No such file or directory
Live child 0x7fb918402e00 (build) PID 3020
Reaping losing child 0x7fb918402e00 PID 3020
make: *** [build] Error 1
```
I had set the env variables in my profile:
```
set GOPATH $HOME/go
set PROMU $GOPATH/src/bin`
```

However, when looking inside the Makefile, I noticed the PROMU variable is set via:
```
    14	GO    := GO15VENDOREXPERIMENT=1 go
    15	PROMU := $(GOPATH)/bin/promu
```

Which was overriding the configured ones in ~/.config/fish/config.fish
The Makefile was already created, so really, consul_exporter just needed to be installed.

My solution was to run 
`go get -u github.com/prometheus/consul_exporter`

This command, "Get downloads the packages named by the import paths, along with their
dependencies. It then installs the named packages, like 'go install'."

Looking in my $GOBIN path (~/go/bin), I can now see the consul_exporter binary installed.
